community/lucene-index/src/main/java/org/neo4j/index/impl/lucene/legacy/LuceneCommandApplier.java

```diff
@@ -72,46 +73,67 @@ public boolean visitIndexAddRelationshipCommand( AddRelationshipCommand command
 
     private boolean visitIndexAddCommand( IndexCommand command, EntityId entityId )
     {
-        CommitContext context = commitContext( command );
-        String key = definitions.getKey( command.getKeyId() );
-        Object value = command.getValue();
-
-        // Below is a check for a null value where such a value is ignored. This may look strange, but the
-        // reason is that there was this bug where adding a null value to an index would be fine and written
-        // into the log as a command, to later fail during application of that command, i.e. here.
-        // There was a fix introduced to throw IllegalArgumentException out to user right away if passing in
-        // null or object that had toString() produce null. Although databases already affected by this would
-        // not be able to recover, which is why this check is here.
-        if ( value != null )
+        try
         {
-            context.ensureWriterInstantiated();
-            context.indexType.addToDocument( context.getDocument( entityId, true ).document, key, value );
+            CommitContext context = commitContext( command );
+            String key = definitions.getKey( command.getKeyId() );
+            Object value = command.getValue();
+
+            // Below is a check for a null value where such a value is ignored. This may look strange, but the
+            // reason is that there was this bug where adding a null value to an index would be fine and written
+            // into the log as a command, to later fail during application of that command, i.e. here.
+            // There was a fix introduced to throw IllegalArgumentException out to user right away if passing in
+            // null or object that had toString() produce null. Although databases already affected by this would
+            // not be able to recover, which is why this check is here.
+            if ( value != null )
+            {
+                context.ensureWriterInstantiated();
+                context.indexType.addToDocument( context.getDocument( entityId, true ).document, key, value );
+            }
+        }
+        catch ( LegacyIndexNotFoundKernelException ignore )
+        {
+            // Pretend the index never existed.
```

Comment:
Useful for recovery, or when could an index just disappear like that?

Reply:
As far as I can tell, there is locking to make legacy index configuration changes mutually exclusive with updates to those indexes, so legacy indexes can basically be dropped at any time.

What happens here is that we ignore updates that were slated for indexes that are dropped concurrently with our commit. Previously, this would have thrown some sort of RuntimeException.

Reply:
I'm absolutely certain you meant to say _...there is NO locking to make legacy index..._. OK then I understand.

Reply:
Yes, that's exactly what I meant to say. Somehow that important little detail didn't make it from my brain to my fingers.